src/NodaTime/TimeZones/IO/TzdbStreamData.cs

```diff
@@ -118,10 +119,16 @@ public DateTimeZone CreateZone(string id, string canonicalId)
         internal static TzdbStreamData FromStream(Stream stream)
         {
             Preconditions.CheckNotNull(stream, nameof(stream));
-            int version = new BinaryReader(stream).ReadInt32();
-            if (version != AcceptedVersion)
+
+            // Using statement to satisfy FxCop, but disposable won't do anything anyway, because
+            // we deliberately leave the stream open.
+            using (var reader = new BinaryReader(stream, Encoding.UTF8, leaveOpen: true))
```

Comment:
... but isn't this actually fixing a bug? Disposal can happen at any time(?), and if `BinaryReader.Dispose()` closes the stream by default (as it seems like it does), then this could have failed future reads, couldn't it?

Reply:
Oh, right, implementing `IDisposable` doesn't imply a finalizer, good point.